cognite_toolkit/_cdf_tk/resource_classes/transformations.py

```diff
@@ -0,0 +1,151 @@
+        "instances",
+        "nodes",
+        "edges",
+        "raw",
+        "sequence_rows",
+    ] = Field(description="The type of the destination resource.")
+
+
+class DataModelSource(DestinationSourceType):
+    data_model: DataModelInfo = Field(description="Target data model info.")
+    instance_space: str | None = Field(default=None, description="The space where the instances will be created.")
+
+
+class ViewDataSource(DestinationSourceType):
+    view: ViewInfo | None = Field(default=None, description="Target view info.")
+    edge_type: EdgeType | None = Field(default=None, description="Target type of the connection definition.")
+    instance_space: str | None = Field(
+        default=None, description="The space where the instances(nodes/edges) will be created."
+    )
+
+
+class RawDataSource(DestinationSourceType):
+    database: str = Field(description="The database name.")
+    table: str = Field(description="The table name.")
+
+
+class SequenceRowDataSource(DestinationSourceType):
+    external_id: str = Field(description="The externalId of sequence.")
+
+
+class AuthenticationClientIdSecret(ToolkitResource):
+    client_id: str = Field(description="Client Id.")
+    client_secret: str = Field(description="Client Secret.")
+    scopes: str | list[str] | None = Field(default=None, description="Scopes for the authentication.")
+    token_uri: str = Field(description="OAuth token url.")
+    cdf_project_name: str = Field(description="CDF project name.")
+    audience: str | None = Field(default=None, description="Audience for the authentication.")
+
+
+class OIDCCredential(ToolkitResource):
+    read: AuthenticationClientIdSecret | None = Field(
+        default=None, description="Source/Read authentication information."
+    )
+    write: AuthenticationClientIdSecret | None = Field(
+        default=None, description="Destination/Write authentication information."
+    )
+
+
+class TransformationYAML(ToolkitResource):
+    external_id: str = Field(description="The external ID provided by the client.")
+    name: str = Field(description="Name of the transformation.")
+    ignore_null_fields: bool = Field(
+        description="Indicates how null values are handled on updates: ignore or set null."
+    )
+    destination: (
+        DestinationSourceType | DataModelSource | ViewDataSource | RawDataSource | SequenceRowDataSource | None
+    ) = Field(default=None, description="Destination data type.")
+    query: str | None = Field(default=None, description="SQL query of the transformation.")
+    conflict_mode: Literal["abort", "delete", "update", "upsert"] | None = Field(
+        default=None,
+        description="Behavior when the data already exists.",
+    )
+    is_public: bool | None = Field(
+        default=None,
+        description="Indicates if the transformation is visible to all in project or only to the owner.",
+    )
+    authentication: AuthenticationClientIdSecret | OIDCCredential | None = Field(
```

Comment:
This can also be (although we lack docs for it)
```suggestion
    authentication: AuthenticationClientIdSecret | OIDCCredential | dict[Literal["read", "write"], AuthenticationClientIdSecret | OIDCCredential]  None = Field(
```